sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -275,10 +275,27 @@ case class Ceil(child: Expression) extends UnaryMathExpression(math.ceil, "CEIL"
       case _ => defineCodeGen(ctx, ev, c => s"(long)(java.lang.Math.${funcName}($c))")
     }
   }
-
   override protected def withNewChildInternal(newChild: Expression): Ceil = copy(child = newChild)
 }
 
+object Ceil {
+  def apply(child: Expression, scale: Expression): Expression = {
+    if (scale.eval(EmptyRow).asInstanceOf[Int] == 0) {
```

Comment:
This throws exception if `scale` is not literal, right?

Reply:
yes, i believe.

Reply:
Can `scale` possibly be evaluated to null and get a NPE here?

Reply:
ah this is a good point. We should return null if scale is null but the question is which return type? This indicates that we should make the behavior change now and let `ceil(c_double, scala)` always return double type. Let's mention this behavior change in `docs/sql-migration-guide.md`